hubblestack/extmods/hubble_mods/osquery.py

```diff
@@ -0,0 +1,217 @@
+        raise HubbleCheckValidationError('Mandatory parameter: {0} not found for id: {1}'.format('query', block_id))
+
+    log.debug('Validation success for check-id: {0}'.format(block_id))
+
+
+def execute(block_id, block_dict, extra_args=None):
+    """
+    Execute the osquery module
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': , 'status': True},
+                  'caller': 'Audit'}
+
+    returns:
+        tuple of result(value) and status(boolean)
+    """
+    log.debug('Executing osquery module for id: {0}'.format(block_id))
+
+    chain_args = extra_args.get('chaining_args')
+    chained_param = runner_utils.get_chained_param(chain_args)
+    # fetch required param
+    query = runner_utils.get_param_for_module(block_id, block_dict, 'query')
+
+    # fetch optional param
+    format_chained = runner_utils.get_param_for_module(block_id, block_dict, 'format_chained', True)
+    osquery_args = runner_utils.get_param_for_module(block_id, block_dict, 'osquery_args')
+    osquery_path = runner_utils.get_param_for_module(block_id, block_dict, 'osquery_path')
+    cast_to_string = runner_utils.get_param_for_module(block_id, block_dict, 'cast_to_string', False)
+
+    if format_chained and chained_param:
+        query = query.format(chained_param)
+    if osquery_args is None:
+        osquery_args = []
+
+    return _osquery(block_id, query, args=osquery_args, osquery_path=osquery_path, cast_to_string=cast_to_string)
+
+
+def get_filtered_params_to_log(block_id, block_dict, extra_args=None):
+    """
+    For getting params to log, in non-verbose logging
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': , 'status': True},
+                  'caller': 'FDG'}
+    """
+    log.debug('get_filtered_params_to_log for id: {0}'.format(block_id))
+
+    chain_args = extra_args.get('chaining_args')
+    # fetch required param
+    chained_param = runner_utils.get_chained_param(chain_args)
+    query = runner_utils.get_param_for_module(block_id, block_dict, 'query')
+    format_chained = runner_utils.get_param_for_module(block_id, block_dict, 'format_chained', True)
+    if format_chained and chained_param:
+        query = query.format(chained_param)
+    return {'query': query}
+
+
+def _osquery(block_id, query, osquery_path=None, args=None, cast_to_string=None):
+    """
+    Format the osquery command and run it
+
+    Returns a tuple, (status, ret) where status is True if the return code is 0,
+    False otherwise, and ``ret`` is the stdout of the osquery command
+    """
+    max_file_size = 104857600
+
+    if not query:
+        return runner_utils.prepare_negative_result_for_module(block_id, '')
```

Comment:
We can put something instead of blank error string. This is at multiple places

Reply:
done